app/handler/books.go

```diff
@@ -112,3 +128,17 @@ func DeleteBook(w http.ResponseWriter, r *http.Request) {
 	books = newSetOfBooks
 	json.NewEncoder(w).Encode(books)
 }
+
+func findOrFail(id string) (book Book, err error) {
+	for _, b := range books {
+		if b.ID == id {
+			book = b
```

Comment:
a good principal is to return early. So, instead of assigning book, just return here, like `return book, nil`

Reply:
a good read on writing good Go code, https://dave.cheney.net/practical-go/presentations/qcon-china.html#_return_early_rather_than_nesting_deeply

Reply:
True but when I `return book, nil` it complains that `have (Book, nil)` `wants (Book, err)`
So I just `return book, err` cause err here will be `<nil>`

Reply:
I will read through this document 🙏